core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -59,6 +61,9 @@
 
     private String performanceCountersSection = DEFAULT_PERFORMANCE_MODULES_PACKAGE;
 
+    final static String ENV_VARIABLE_I_KEY = "APPLICATION_INSIGHTS_IKEY";
+    final static String JVM_VARIABLE_I_KEY = "-D" + ENV_VARIABLE_I_KEY;
```

Comment:
Are you sure that this would be the name of the variable provided by the JVM?
Per my understanding when java is executed with a flag `-Dproperty=value` then the property name is `property` rather than `-Dproperty`